lib/server.js

```diff
@@ -312,20 +648,637 @@ function createServer(options, clients) {
+      *  - List buckets
+      *  - List bucket contents
+      *  - Check if Bucket exists
+      *  - Create Bucket
+      *  - Delete Bucket (here post is a special case when bulk deleting)
+      */
+    server.get(/^\/([^\/]+)$/, function (req, res, next) {
+        if (req.isS3Request) {
+            req.log.debug('S3_DEBUG: S3 single-segment GET route'+
+            ' handling S3 request');
+            s3RouteHandler(req, res, next);
+        } else {
+            req.log.debug('S3_DEBUG: S3 single-segment GET route '+
+            'passing through non-S3 request');
+            next();
+        }
+    });
+
+    server.put(/^\/([^\/]+)$/, function (req, res, next) {
+        if (req.isS3Request) {
+            s3RouteHandler(req, res, next);
+        } else {
+            next();
+        }
+    });
+
+    server.head(/^\/([^\/]+)$/, function (req, res, next) {
+        if (req.isS3Request) {
+            s3RouteHandler(req, res, next);
+        } else {
+            next();
+        }
+    });
+
+    server.del(/^\/([^\/]+)$/, function (req, res, next) {
+        if (req.isS3Request) {
+            s3RouteHandler(req, res, next);
+        } else {
+            next();
+        }
+    });
+
+    server.post(/^\/([^\/]+)$/, function (req, res, next) {
+        if (req.isS3Request) {
+            req.log.debug('S3_DEBUG: S3 single-segment POST' +
+                          ' route handling S3 request');
+            s3RouteHandler(req, res, next);
+        } else {
+            req.log.debug('S3_DEBUG: S3 single-segment POST' +
+                          ' route passing through non-S3 request');
+            next();
+        }
+    });
+
+     /*
+      * One or more path segment Bucket Operations for S3 and Manta bucket
+      * objects
+      *
+      *  The regex route /^\/([^\/]+)\/(.+)$/ catches both:
+      *  - S3 requests: /bucket/object
+      *  - Manta requests: /account/buckets/bucket_name/objects/object_name
+      *
+      * What does it means?
+      * It just means the request has the following form:
+      *  For S3 requests:
+      *  - /bucket/object
+      *  - /bucket/somepath/object
+      *  For Manta requests:
+      *  - /account/buckets/bucket_name/objects/object_name
+      *
+      * Here we are directly operating on a bucket object
+      * and those operations are : GET, HEAD, PUT, DELETE
+      * Which translates to :
+      *  - List bucket contents
+      *  - Check if object exists
+      *  - Create Object in Bucket
+      *  - Delete Object in Bucket
+      */
+
+    server.get(/^\/([^\/]+)\/(.+)$/, function (req, res, next) {
+        req.log.debug({
+            path: req.path(),
+            isS3Request: !!req.isS3Request
+        }, 'S3_DEBUG: GET two-segment route - checking if S3 request');
+
+        if (req.isS3Request) {
+            req.log.debug('S3_DEBUG: Processing as S3 request'+
+            ' in GET two-segment route');
+            s3RouteHandler(req, res, next);
+        } else {
```

Comment:
Same indentation idea here?

```js
if (req.isS3Request) {
    req.log.debug('S3_DEBUG: Processing as S3 request'+
            ' in GET two-segment route');
    s3RouteHandler(req, res, next);
    return;
}

// Rest is non-S3 specific

/*
 * We should really process the traditional Manta route but,
 * as we allowed anonymous object access through http
 * that falls into a Manta route so we need to execute more steps
 * before falling through, only if the object has anonymous access.
 */
```

Reply:
I fixed the indentation problems from server.js, and I'll be re-visiting the rest of the files.

Reply:
Will need to revisit these on a next pass.  Resolving this for now.